### PR TITLE
Add a README.md on dind with minimal information

### PR DESCRIPTION
**Pull request by manuelbuil**
I tried today DinD and it was a bit hard to find information about how to run it. I dumped what I learnt in a README.md

### CODE REVIEW

File: dind/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# rke DinD
+
+Docker-in-Docker (DinD) feature is created mainly for testing.
+
+DinD creates docker containers locally which are then used as cluster nodes.
+
+It still requires a cluster.yml with minimal information:
+
+```
+kubernetes_version: "$KUBE_VERSION"
```
Comment (superseb) [inline dind/README.md]: Either leave this out (as it is minimal) or specify an example so users know what to put here

File: dind/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+nodes:
+- address: foo
```
Comment (superseb) [inline dind/README.md]: Possibly add this needs to be unique even though it's a dummy

File: dind/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  role:
+  - controlplane
+  - worker
+  - etcd
+  user: bar
+```
+
+To create a cluster with DinD execute:
+`docker up --dind`
```
Comment (superseb) [inline dind/README.md]: `rke up`

File: dind/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+After the creation is completed, you will see one docker container per node running. You can use the created `kube_config_cluster.yml` to access the cluster as in a normal rke deployment
+
+To remove a cluster with DinD execute:
+`docker remove --dind`
```
Comment (superseb) [inline dind/README.md]: `rke remove`

File: dind/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+# FAQ
+
+* Do I need to create a network for the docker containers?
+
+No. The created docker containers use the default docker bridge network which allows to connect from the host to the containers on IPs in the subnet "172.17.0.0/16". However, they have no ssh service running, therefore you `docker exec` if you want to connect to them
+
+* What image is used for those containers?
+
+There is a special image: rancher/docker:19.03.12-dind that is used for that
```
Comment (superseb) [inline dind/README.md]: Where does this come from? We are at 20.10.23 (https://github.com/rancher/rke/blob/release/v1.5/dind/dind.go#L16) and it's not a special image, it is from upstream (https://hub.docker.com/_/docker)

**Comment by superseb** [comment]
In general, Docker and Kubernetes are brands/products and should be capitalized. RKE is product, `rke` is CLI usage.

Also, please move this to `/docs`

File: dind/README.md (hunk 3 above)
Comment (manuelbuil) [reply dind/README.md]: thanks, good catch!

File: dind/README.md (hunk 4 above)
Comment (manuelbuil) [reply dind/README.md]: thanks, good catch!

File: dind/README.md (hunk 5 above)
Comment (manuelbuil) [reply dind/README.md]: When using rke from release 1.4.10, I see in the logs:
```
INFO[0000] Running RKE version: v1.4.10                 
INFO[0000] Pulling image [docker:19.03.12-dind] on host [unix:///var/run/docker.sock], try #1 
INFO[0017] Image [docker:19.03.12-dind] exists on host [unix:///var/run/docker.sock] 
```

File: dind/README.md (hunk 1 above)
Comment (manuelbuil) [reply dind/README.md]: I wasted a lot of time trying to understand what needs to be passed to a cluster.yaml file of a dind deployment, so I'd prefer to leave it. I can specify how to find `$KUBE_VERSION`

File: dind/README.md (hunk 2 above)
Comment (manuelbuil) [reply dind/README.md]: ok

File: dind/README.md (hunk 5 above)
Comment (superseb) [reply dind/README.md]: This PR is on the `release/v1.5` branch which uses a different version. I would remove the version and link users to the file where the image is specified so they can look it up themselves.

It's still not a special image, its the upstream Docker image.

File: docs/dind.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+# FAQ
+
+* Do I need to create a network for the docker containers?
+
+No. The created docker containers use the default docker bridge network which allows to connect from the host to the containers on IPs in the subnet "172.17.0.0/16". However, they have no ssh service running, therefore you `docker exec` if you want to connect to them
+
+* What image is used for those containers?
+
+The default docker image mirrores in rancher registry: rancher/docker:${VERSION}-dind that is used for that
```
Comment (superseb) [inline docs/dind.md]: Please share where you got this from, the image is not mirrored to rancher or coming from Rancher. It is upstream's Docker image.

**Comment by superseb** [comment]
Docker/Kubernetes are not capitalized still, I'm fine with approving if we get clarity on the Docker image being used for DinD

**Comment by manuelbuil** [comment]
> Docker/Kubernetes are not capitalized still, I'm fine with approving if we get clarity on the Docker image being used for DinD

Sorry Seb, I'm stupid. Let me capitalize them

**Review by superseb — APPROVED** [review]

**Review by thomasferrandiz — APPROVED** [review]

**Pull request merged**